src/utils/codemods/findAvailableCodemodsInNodeModules.js

```diff
@@ -2,32 +2,81 @@ const log = require('@dhis2/cli-helpers-engine').reporter
 const fs = require('fs-extra')
 const path = require('path')
 
-const isNotDotFolder = value => value !== '.' && value !== '..'
+const isDotFolder = name => name[0] === '.'
+
+const isNotDotFolder = name => !isDotFolder(name)
 
 const isDotJSFile = name => new RegExp('[.]js$').test(name)
 
-const extractCodemods = (paths, moduleName) => {
-    const { DHIS2_NODE_MODULES } = paths
-    const codemodsPath = path.join(DHIS2_NODE_MODULES, moduleName, 'codemods')
+const isFolder = path => {
+    if (!fs.existsSync(path)) return false
+    return fs.lstatSync(path).isDirectory()
+}
 
-    if (!fs.pathExistsSync(codemodsPath)) {
-        log.debug(
-            `Cound not find a codemods folder in node module '${moduleName}' directory`
-        )
+const hasCodemods = codemodsPath =>
+    !!fs.readdirSync(codemodsPath).filter(curFile => {
+        const curPath = path.join(codemodsPath, curFile)
 
-        return []
-    }
+        if (isFolder(curPath)) return false
+        if (!curFile.match(/\.(j|t)sx?$/)) return false
+
+        return true
+    }).length
+
+/**
+ * Tries to find the following dirs that contain .js files:
+ *   - ${NODE_MODULES}/{{folder}}/codemods/
+ *   - ${NODE_MODULES}/{{namespace}}/{{folder}}/codemods/
+ *
+ * @param {Object} paths
+ * @returns {String[]}
+ */
+const findCodemodsFolders = paths => {
+    return fs
+        .readdirSync(paths.NODE_MODULES)
```

Comment:
As an FYI this will break completely when using I.e. yarn2 or pnpm since there's no `node_modules`.  It would be better to look at all dependencies and devDependencies and look for code mods in their virtual FS wherever that happens to be

Reply:
We can do this in a separate change if needed

Reply:
I will try to get that in today or tomorrow after merging this one, not sure if I can manage though, I first have to figure out how to access the virtual FS, haven't touched yarn2 yet)